ggshield/verticals/secret/output/secret_json_output_handler.py

```diff
@@ -27,7 +27,7 @@ def create_scan_dict(
         if scan.extra_info:
             scan_dict["extra_info"] = scan.extra_info
 
-        if top and scan.has_results:
+        if top:
```

Comment:
I am not sure why the original code behaved that way, but is there any reason for this change?

Reply:
Yes, I removed the `has_result` attribute of `ScanCollection`, which wasn't used anywhere else (and was a bit bugged, since it did not consider sub-scans)